backend/tracim_backend/applications/upload_permissions/lib.py

```diff
@@ -0,0 +1,284 @@
+    UploadPermissionEmailManager,
+)
+from tracim_backend.applications.upload_permissions.models import UploadPermission
+from tracim_backend.applications.upload_permissions.models import UploadPermissionType
+from tracim_backend.applications.upload_permissions.models_in_context import (
+    UploadPermissionInContext,
+)
+from tracim_backend.config import CFG
+from tracim_backend.exceptions import NotificationSendingFailed
+from tracim_backend.exceptions import UploadPermissionNotFound
+from tracim_backend.exceptions import WrongSharePassword
+from tracim_backend.lib.core.content import ContentApi
+from tracim_backend.lib.mail_notifier.utils import SmtpConfiguration
+from tracim_backend.lib.utils.logger import logger
+from tracim_backend.lib.utils.utils import get_frontend_ui_base_url
+from tracim_backend.models.auth import User
+from tracim_backend.models.context_models import ContentInContext
+from tracim_backend.models.data import ActionDescription
+from tracim_backend.models.data import ContentNamespaces
+from tracim_backend.models.data import Workspace
+from tracim_backend.models.revision_protection import new_revision
+
+FRONTEND_UPLOAD_PERMISSION_LINK_PATTERN = (
+    "{frontend_ui_base_url}guest-upload/{upload_permission_token}"
+)
+
+
+class UploadPermissionLib(object):
+    def __init__(
+        self,
+        session: Session,
+        current_user: typing.Optional[User],
+        config: CFG,
+        show_disabled: bool = False,
+    ) -> None:
+        self._user = current_user
+        self._session = session
+        self._config = config
+        self.show_disabled = show_disabled
+
+    def base_query(self) -> Query:
+        base_query = self._session.query(UploadPermission)
+        if not self.show_disabled:
+            base_query = base_query.filter(UploadPermission.enabled == True)  # noqa: E712
+        return base_query
+
+    def add_permission_to_workspace(
+        self,
+        workspace: Workspace,
+        emails: typing.List[str],
+        password: typing.Optional[str] = None,
+        do_notify=True,
+    ) -> typing.List[UploadPermission]:
+        upload_permissions = []
+        created = datetime.now()
+        upload_permission_group_uuid = str(uuid.uuid4().hex)
+        for email in emails:
+            upload_permission = UploadPermission(
+                author=self._user,
+                workspace_id=workspace.workspace_id,
+                email=email,
+                token=str(uuid.uuid4()),
+                password=password,
+                type=UploadPermissionType.EMAIL,
+                created=created,
+                upload_permission_group_uuid=upload_permission_group_uuid,
+                enabled=True,
+            )
+            self.save(upload_permission)
+            upload_permissions.append(upload_permission)
+            self._session.flush()
+
+        if do_notify and self._config.EMAIL__NOTIFICATION__ACTIVATED:
```

Comment:
If `do_notify`is True and `self._config.EMAIL__NOTIFICATION__ACTIVATED` is False, we must have an error (maube not and fatal error, but at least a log)

Reply:
code changed to not have this kind of case.